### DISCUSSION TITLE
How fuse two graph with different nodes in attention?

### DISCUSSION THREAD

**Discussion by fengsxy** [pyg-team/pytorch_geometric#4735] [Q&A]
While doing regression task,i would like to use two different graph to interact and trian but i find it is difficult to fuse two graph with different nodes.Do we have some methods to do this?For example ,i have two graph ,data_1.x is (1300,100) data_2.x is(800,100),how can i 
fuse it ?

**Comment by rusty1s** [comment]
You can use the `torch_geometric.loader.DataLoader` to do this:
```python
from torch_geometric.loader import DataLoader
loader = DataLoader([data1, data2], batch_size=2, ...)
batch = next(iter(loader))
```
The `DataLoader` will call `Batch.from_data_list` behind the scenes. The above call is therefore equivalent to:
```python
from torch_geometric.data import Batch
batch = Batch.from_data_list([data1, data2])
```

**Reply by fengsxy** [reply]
Thank you for your answer,I handle like this,
```python
    dataset_1 = process.get_dataset(data_path,_1 training_parameters["target_index"], False)
    dataset_2 = process.get_dataset(data_path_2)
    dataset = MyDataset(dataset_1,dataset_2)
class MyDataset(Dataset):
    # 默认预处理函数的参数都是None
    def __init__(self, Dataset_1, Dataset_2):
        super().__init__()
        self.Dataset_1 = Dataset_1
        self.Dataset_2 = Dataset_2
    def len(self):
        return len(self.Dataset_1)

    def get(self, idx):
        return self.Dataset_1[idx],self.Dataset_2[idx]
```

**Reply by fengsxy** [reply]
And i have the question about algorthm,when we have two graph with different nodes,but we want make them map in the same node,do we have any ways to achieve this.May be this question is awkward,i am trying to find some solution now!

**Reply by rusty1s** [reply]
Ah, then I misunderstood your question. Sorry for that. Your above solution looks good to me. What do you mean with mapping them to the same node?

**Reply by fengsxy** [reply]
For exampe,i have two graphs,the first have 100 nodes ,feature_channel is 15 ,it's like 100 * 15,
i have the other graph,it has 200 nodes ,feature channel is 15, Now i want this two graph can be interacted,can i use pool .The type of interacting has a lot of ways.
For example ,i can just pool the second graph to change it from 200 Nodes to 100 Nodes.Make them have the same nodes,and then they can simply submission.And do we have some tricks like this to make two different nodes graph to infect with each other？This is a question which disturbs me a lot of times,i am trying to find some typical or classical ways.
Because in the Euclidean space.if two images are different ,we can reszie ,Downsample to make them similar. In non-Euclidean,if it is possible to have some ways like the Downsample.
I am trying to explore this too,if you have any ideas about ,i would really appreaciate about it!

**Reply by rusty1s** [reply]
I think this depends on the task you want to do. In general, you cannot simply re-size a graph. In addition, the graph does not have a fixed ordering such that node 1 in graph 1 corresponds to node 1 in graph 2. If you want to let the two graphs exchange information, you can think about adding inter-edges between those two graphs. For example, by constructing a k-NN graph via `torch_cluster.knn` in the feature space. Let me know if this may fit your needs!

**Reply by fengsxy** [reply]
Really,thank u! I think this idea is fantastic.I would try it.
